### PR TITLE
DT-727 gradle plugin

### PR DESCRIPTION
**Pull request by mikehalmamoj**
Add git properties plugin.  Add Spring Boot Test and exclude Junit 4.

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/JavaBuildFuncTest.kt [hunk 1]
```diff
@@ -12,22 +13,18 @@ import java.util.jar.JarFile
...
-
-    @TempDir
-    @JvmStatic
-    lateinit var projectDir: File
-
-    @BeforeAll
-    @JvmStatic
-    fun `Create and run project`() {
-      makeProject(javaProjectDetails(projectDir))
-    }
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/JavaBuildFuncTest.kt:25]: The run once approach is good when spinning up the Spring Boot project but doesn't make sense when just testing the Gradle build.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/JavaBuildFuncTest.kt [hunk 2]
```diff
@@ -72,11 +77,41 @@ class JavaBuildFuncTest {
...
+  fun `Junit 5 tests can be executed`() {
+    makeProject(javaProjectDetails(projectDir))
+
+    val result = buildProject(projectDir, "test")
+    assertThat(result.task(":test")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+  }
+
+  @Test
+  fun `Junit 4 tests will not event compile`() {
+    makeProject(javaProjectDetails(projectDir).copy(testClass = """
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/JavaBuildFuncTest.kt:99]: As the project is refreshed each time I can create a Junit 4 test class to prove it does not work.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/DpsSpringBootPlugin.kt [hunk 3]
```diff
@@ -41,6 +42,7 @@ class DpsSpringBootPlugin : Plugin<Project> {
     addDependencyCheckSuppressionFile(project)
     rejectUnstableDependencyUpdates(project)
     setKotlinCompileJvmVersion(project)
+    excludeJunit4(project)
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/DpsSpringBootPlugin.kt:45]: what does this do? Sounds scary?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/JavaBuildFuncTest.kt (hunk 2 above)
Comment (andymarke) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/JavaBuildFuncTest.kt:99]: hmm? Not sure I understand, are saying all tests *must* be Junit 5? If so why?

**Pull request merged**